InformationScripting/scripts/color.py

```diff
@@ -8,6 +8,6 @@
 for ts in inputs:
     tuples = ts.take("ast")
     for tuple in tuples:
-        tuple.add(NamedProperty("color", color))
-        ts.add(tuple)
+        t = Tuple([NamedProperty("color", color), NamedProperty("ast", tuple.ast)])
```

Comment:
Since type checking is dynamic, the relevant question is at what point can we detect a type error in the current case vs. tuples. If we can still do it before the assignment to `t` in the line above then that sounds good to me.

BTW, wouldn't it make even more sense to initialize `Tuple` not with a list of things, but with a dictionary? That way we're guaranteed to have two parts per element?